src/backend/joanie/core/admin.py

```diff
@@ -460,6 +467,17 @@ def generate_certificates(self, request, queryset):  # pylint: disable=no-self-u
 
         summarize_certification_to_user(request, certificate_generated_count)
 
+    def publish_on_course(self, request, product_id, course_code):
+        """
+        Custom action to publish product on the related course
+        """
+        options = {"course": course_code, "product": product_id}
+        call_command("publish_course_run", **options)
```

Comment:
What about using a celery task here ? Can we return a response before the end of this command ?

Reply:
We could queued the webhook logic and return a response as soon as the webhook call has been queued